### PR TITLE
Change how number of pending messages is calculated and add more error handling.

### PR DESCRIPTION
**Pull request by cwoolum**
Makes a change for #520 and adds additional logging to help diagnose connection issues.

### CODE REVIEW

File: pkg/scalers/stan_scaler.go [hunk 1]
```diff
@@ -108,30 +108,51 @@ func parseStanMetadata(metadata map[string]string) (stanMetadata, error) {
...
+}
+
 func (s *stanScaler) getMonitoringEndpoint() string {
-	return "http://" + s.metadata.natsServerMonitoringEndpoint + "/streaming/channelsz?" + "channel=" + s.metadata.subject + "&subs=1"
+	return s.getSTANChannelsEndpoint() + "?channel=" + s.metadata.subject + "&subs=1"
 }
 
 func (s *stanScaler) getTotalMessages() int64 {
-	return s.channelInfo.MsgCount
+	return s.channelInfo.LastSequence
```
Comment (ahmelsayed) [inline pkg/scalers/stan_scaler.go]: Looking at the code, I don't think this method is actually used anywhere, right? 

I think `getMaxMsgLag()` below calls `return s.channelInfo.MsgCount - maxValue` and that's probably the part @balchua meant needs to be changed if I'm not mistaken. 

Sorry if I misunderstood the issue. I'm trying to catch up on few things.

**Comment by ahmelsayed** [comment]
@balchua it would be great if you can take a look at this PR whenever you get the time :)

**Comment by balchua** [comment]
@ahmelsayed yup on it. You got that right.  Should have used the `getTotalMessages` method.

File: pkg/scalers/stan_scaler.go (hunk 1 above)
Comment (cwoolum) [reply pkg/scalers/stan_scaler.go]: You are right. I updated the incorrect line. `getTotalMessages` is never called so I removed it this time.

File: pkg/scalers/stan_scaler.go [hunk 2]
```diff
@@ -108,30 +108,51 @@ func parseStanMetadata(metadata map[string]string) (stanMetadata, error) {
...
-	maxValue = 0
+	maxValue := int64(0)
+	combinedQueueName := s.metadata.durableName + ":" + s.metadata.queueGroup
+
 	for _, subs := range s.channelInfo.Subscriber {
-		if subs.LastSent > maxValue && subs.QueueName == (s.metadata.durableName+":"+s.metadata.queueGroup) {
+		if subs.LastSent > maxValue && subs.QueueName == combinedQueueName {
 			maxValue = subs.LastSent
 		}
 	}
```
[the remark this answers is not in the document]
Comment (cwoolum) [reply pkg/scalers/stan_scaler.go]: Is there anything you would like me to change for this or am I fine removing the `getTotalMessages()` method?

[the remark this answers is not in the document]
Comment (balchua) [reply pkg/scalers/stan_scaler.go]: Its fine to take out that method like you did.  It is not used anywhere anyway.  The PR looks good! Thanks!

**Comment by ahmelsayed** [comment]
Thank you @cwoolum  and @balchua for the review :)

**Comment by balchua** [comment]
@ahmelsayed the PR LGTM.

**Pull request merged**